components/board/upboard/data.go

```diff
@@ -15,40 +15,41 @@ var boardInfoMappings = map[string]genericlinux.BoardInformation{
 				GPIOChipSysFSDir: path to the directory of a chip. Can be found from the output of gpiodetect
 			*/
 			// GPIO pin definition
-			{Name: "29", Ngpio: 78, LineNumber: 73, PwmChipSysfsDir: "", PwmID: -1},
-			{Name: "31", Ngpio: 77, LineNumber: 46, PwmChipSysfsDir: "", PwmID: -1},
-			{Name: "18", Ngpio: 77, LineNumber: 48, PwmChipSysfsDir: "", PwmID: -1},
-			{Name: "22", Ngpio: 77, LineNumber: 45, PwmChipSysfsDir: "", PwmID: -1},
-			{Name: "37", Ngpio: 77, LineNumber: 46, PwmChipSysfsDir: "", PwmID: -1},
-			{Name: "35", Ngpio: 47, LineNumber: 17, PwmChipSysfsDir: "", PwmID: -1},
-			{Name: "13", Ngpio: 77, LineNumber: 75, PwmChipSysfsDir: "", PwmID: -1},
+			{Name: "29", Ngpio: 28, LineNumber: 5, PwmChipSysfsDir: "", PwmID: -1},
+			{Name: "31", Ngpio: 28, LineNumber: 6, PwmChipSysfsDir: "", PwmID: -1},
+			{Name: "18", Ngpio: 28, LineNumber: 24, PwmChipSysfsDir: "", PwmID: -1},
+			{Name: "22", Ngpio: 28, LineNumber: 25, PwmChipSysfsDir: "", PwmID: -1},
+			{Name: "37", Ngpio: 28, LineNumber: 26, PwmChipSysfsDir: "", PwmID: -1},
+			{Name: "35", Ngpio: 28, LineNumber: 19, PwmChipSysfsDir: "", PwmID: -1},
+			{Name: "13", Ngpio: 28, LineNumber: 27, PwmChipSysfsDir: "", PwmID: -1},
 
 			// ttyS4 UART
-			{Name: "8", Ngpio: 78, LineNumber: 43, PwmChipSysfsDir: "", PwmID: -1},
-			{Name: "10", Ngpio: 78, LineNumber: 42, PwmChipSysfsDir: "", PwmID: -1},
-			{Name: "11", Ngpio: 78, LineNumber: 44, PwmChipSysfsDir: "", PwmID: -1},
-			{Name: "36", Ngpio: 78, LineNumber: 45, PwmChipSysfsDir: "", PwmID: -1},
+			{Name: "8", Ngpio: 28, LineNumber: 14, PwmChipSysfsDir: "", PwmID: -1},
+			{Name: "10", Ngpio: 28, LineNumber: 15, PwmChipSysfsDir: "", PwmID: -1},
+			{Name: "11", Ngpio: 28, LineNumber: 17, PwmChipSysfsDir: "", PwmID: -1},
+			{Name: "36", Ngpio: 28, LineNumber: 16, PwmChipSysfsDir: "", PwmID: -1},
 
 			// I2c
-			{Name: "3", Ngpio: 78, LineNumber: 28, PwmChipSysfsDir: "", PwmID: -1},
-			{Name: "5", Ngpio: 78, LineNumber: 29, PwmChipSysfsDir: "", PwmID: -1},
-			{Name: "28", Ngpio: 78, LineNumber: 31, PwmChipSysfsDir: "", PwmID: -1},
+			{Name: "3", Ngpio: 28, LineNumber: 2, PwmChipSysfsDir: "", PwmID: -1},
+			{Name: "5", Ngpio: 28, LineNumber: 3, PwmChipSysfsDir: "", PwmID: -1},
+			{Name: "28", Ngpio: 28, LineNumber: 1, PwmChipSysfsDir: "", PwmID: -1},
```

Comment:
Off-topic question: why are there 3 pins in the I2C section? I'd expect either 2 (for a single I2C bus) or 4 (for a second bus).

Reply:
I'll sort them out by name. I didn't really do much here other than replacing number.